src/io/github/sspanak/tt9/ime/TraditionalT9.java

```diff
@@ -349,14 +349,17 @@ public boolean onKeyAddWord() {
 
 	public boolean onKeyNextLanguage() {
 		if (nextLang()) {
-			commitCurrentSuggestion(false);
 			mInputMode.changeLanguage(mLanguage);
-			mInputMode.reset();
+			mInputMode.clearWordStem();
+			getSuggestions();
 			resetKeyRepeat();
```

Comment:
```suggestion
			clearSuggestions();
			getSuggestions();
			resetKeyRepeat();
```

`clearSuggestions()` + `getSuggestions()` is the equivalent of what you tried to do in `setInputWord()`. Let's do it this way.